src/auth/strategies/magic-link.strategy.ts

```diff
@@ -26,18 +26,27 @@ export class MagicLinkStrategy extends PassportStrategy(
   async authenticate(req: Request): Promise<void> {
     const { authorization } = req.headers;
     if (!authorization) {
-      return this.fail('unauthorized');
+      return this.fail({
+        code: 'no_authorization',
+        message: 'Error: Request header is invalid.',
```

Comment:
nit: If we're checking `!authorization`, then the header is undefined or `''`

```suggestion
        message: 'Error: Request header is missing.',
```

Reply:
The request header is not missing, the request header is missing authorization data. That is why I put invalid request header. But I dont think we need to be specific to user about the authorization data is missing.